### PR TITLE
fix KubePodProcess port offering race condition

### PR DESCRIPTION
**Pull request by jrhizor**
There's a race condition when we `offer` a port back after closing a `KubePodProcess` since the unique. check of the offering isn't atomic:
```
    if (!workerPorts.contains(port)) {
      workerPorts.add(port);
    }
```

The problem happens where we `close()` multiple times, calling the above code multiple times. It's possible for the port to already be picked up (so the contains check allows it to add a port) when the port was already consumed by a different process.

This tests for the prior case and and fixes the issue. 

This issue would manifest by one pod process failing before causing another to fail due to the same address already being in use. This shows up rarely in cloud logs (but we would have expected it to show up much more often before this fix).

### CODE REVIEW

File: airbyte-workers/src/main/java/io/airbyte/workers/process/KubePodProcess.java [hunk 1]
```diff
@@ -517,6 +518,14 @@ public Info info() {
    * implementation with OS processes and resources, which are automatically reaped by the OS.
    */
   private void close() {
+    // short-circuit if close was already called, so we don't re-offer ports multiple times
```
Comment (davinchia) [inline airbyte-workers/src/main/java/io/airbyte/workers/process/KubePodProcess.java:521]: should this be a `synchronised` method?

Comment (davinchia) [reply airbyte-workers/src/main/java/io/airbyte/workers/process/KubePodProcess.java:521]: or the boolean by an atomic boolean?

Comment (davinchia) [reply airbyte-workers/src/main/java/io/airbyte/workers/process/KubePodProcess.java:521]: I think atomic is preferred as it's hardware level while synchronise is a JVM wide locking mechanims.

**Review by davinchia — COMMENTED** [review]
Nice!

One blocking comment to make sure close itself is run exactly once.

**Comment by jrhizor** [comment]
Passing everything except logging tests which are broken due to secrets removed from Github.

**Review by davinchia — APPROVED** [review]

**Pull request merged**